src/dal_select2/static/autocomplete_light/select2.js

```diff
@@ -8,13 +8,24 @@
             var data_forward = {};
 
             for (var key in forward) {
+                var f = forward[key].split("->");
+                var src_name, dst_name;
+                if (f.length === 0) {
+                    continue;
+                } else if (f.length === 1) {
+                    src_name = f[0];
+                    dst_name = f[0];
+                } else {
+                    src_name = f.splice(0,1);
+                    dst_name = f.join("->");
+                }
```

Comment:
That looks pretty cool indeed, we could also have a js callback type.

I know that storing json in html data attributes doesn't work very well
indeed. However, we shouldn't have that script tag inside the select field
because we want to support JAL which uses an input tag instead of a select.

Do you think it will work to have the script tag after the select tag ?

On Sep 11, 2016 1:20 PM, "Kirill Gagarski" <notifications@github.com> wrote:

> In src/dal_select2/static/autocomplete_light/select2.js
> <https://github.com/yourlabs/django-autocomplete-light/pull/733#discussion_r78292847>
> :
>
> > @@ -8,13 +8,24 @@
> >              var data_forward = {};
> >
> >              for (var key in forward) {
> > +                var f = forward[key].split("->");
> > +                var src_name, dst_name;
> > +                if (f.length === 0) {
> > +                    continue;
> > +                } else if (f.length === 1) {
> > +                    src_name = f[0];
> > +                    dst_name = f[0];
> > +                } else {
> > +                    src_name = f.splice(0,1);
> > +                    dst_name = f.join("->");
> > +                }
>
> We can use JSON but we probably should not store it in an HTML attribute
> because it involves ugly escaping/unescaping code.
> A possible way to handle this is something like that:
>
> <select><!--with some select2 and dal attrs-->
>   <!-- Here we declare three forwards:    - forward field_name value without renaming    - forward field_name2 value as forward_name    - forward constant 42 as const_forward_name  -->
>   <script type="text/forwardconfig">
>     [
>       {
>         "type": "field",
>         "src": "field_name"
>       },
>       {
>         "type": "field",
>         "src": "field_name2",
>         "dst": "forward_name"
>       },
>       {
>         "type": "const",
>         "value": 42
>         "dst": "const_forward_name"
>       }
>     ]
>   </script>
>   <!-- Select options -->
> </select>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/yourlabs/django-autocomplete-light/pull/733/files/f68ca745ad7c2417405eba85b06bf1453eb139bf#r78292847>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFxrECOClwCYWXJsp3UITDPMgbQ4NKEks5qo-PpgaJpZM4JzLkD>
> .
>